ipatests/test_integration/test_trust_functional.py

```diff
@@ -304,3 +308,355 @@ def test_ipa_trust_func_hbac_0011(self):
         finally:
             self._cleanup_hrule_allow_all_and_wait(hrule)
             self.master.run_command(["ipa", "sudorule-del", srule])
+
+
+class TestTrustFunctionalSudo(BaseTestTrust):
+    topology = 'line'
+    num_ad_treedomains = 0
+
+    def cache_reset(self):
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.clients[0])
+        tasks.wait_for_sssd_domain_status_online(self.master)
+        tasks.wait_for_sssd_domain_status_online(self.clients[0])
+        time.sleep(30)
+
+    def _cleanup_srule(self, srule):
+        tasks.kinit_admin(self.master)
+        self.master.run_command(["ipa", "sudorule-del", srule])
+        self.cache_reset()
+
+    def _run_sudo_command(self, host, command, username, password='Secret123',
+                          expected_output=None, raiseonerr=True, timeout=60):
+        """
+        Run a sudo command using spawn_expect with proper error handling.
+
+        Args:
+            host: Host to run the command on
+            command: Command to execute
+            username: Username for password prompt
+            password: Password to send
+            expected_output: Expected string in output (for assertion)
+            raiseonerr: Whether to raise on error
+            timeout: Timeout for expect_exit
+
+        Returns:
+            Output from the command
+        """
+        with host.spawn_expect(command) as e:
+            e.expect(r'.*assword for {}.*:'.format(username))
+            e.sendline(password)
+            e.expect_exit(ignore_remaining_output=True,
+                          raiseonerr=raiseonerr, timeout=timeout)
+            output = e.get_last_output()
+
+            if expected_output:
+                assert expected_output in output, (
+                    f"Expected '{expected_output}' in output, got: {output}"
+                )
+
+            return output
+
+    def test_ipa_trust_func_sudo_setup(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+        tasks.kinit_admin(self.master)
+
+        for i in range(1, 4):
```

Comment:
`for i in range(1, 3):` I think this range is enough, you're not using `sudogroup_external3` nor `sudogroup3`